File: laboratories/ppo/exercises/lab07/python/src/vehicles/Car.py
from src.vehicles.Vehicle import Vehicle


class Car(Vehicle):
    plates = None

    def __init__(self, plates, name = None):
        if name:
           super().__init__(name)
        else:
           super().__init__()

        self.plates = plates

    def _get_identifier(self):
        return self.identifier + " by car [" + self.plates + "]"

    def _get_anonymous_identifier(self):
        return "anonymous by car [" + self.plates + "]"

File: laboratories/ppo/exercises/lab07/php/src/Vehicles/Car.php
<?php

namespace Ppo\Lab07\Vehicles;

class Car extends Vehicle
{
    protected string $plates;

    public function __construct(string $plates, ?string $name = null)
    {
        parent::__construct($name);
        $this->plates = $plates;
    }

    protected function getIdentifier(): string
    {
        return "{$this->identifier} by car [{$this->plates}]";
    }

    protected function getAnonymousIdentifier(): string
    {
        return "anonymous by car [{$this->plates}]";
    }
}
